custom_components/ha-google-home/api.py

```diff
@@ -52,62 +54,88 @@ def get_android_id(self):
-        response = requests.get(url, headers=HEADERS, verify=False, timeout=TIMEOUT)
-
-        if response.status_code != HTTP_OK:
-            _LOGGER.error(
-                "For device {device} - API returned {error}: {reason}".format(
-                    device=device.device_name,
-                    error=response.status_code,
-                    reason=response.text,
-                )
+        url = self.create_url(device.ip, PORT, endpoint)
+        if url:
+            _LOGGER.debug(
+                "For device {device} - {url}".format(device=device.device_name, url=url)
             )
-            return
+            # verify=False is need to avoid SSL security checks. Otherwise it will fail to connect"""
+            HEADERS[HEADER_CAST_LOCAL_AUTH] = device.local_auth_token
+            response = requests.get(url, headers=HEADERS, verify=False, timeout=TIMEOUT)
+
+            if response.status_code != HTTP_OK:
+                _LOGGER.error(
+                    "For device {device} - API returned {error}: {reason}".format(
+                        device=device.device_name,
+                        error=response.status_code,
+                        reason=response.text,
+                    )
+                )
+                return
+            else:
+                return response.json()
         else:
-            return response.json()
+            return
 
     def get_google_devices_information(self):
         """Retrieves devices from glocaltokens"""
         _LOGGER.debug("Fetching data...")
+        offline_devices = []
         devices = self._client.get_google_devices()
 
         for device in devices:
             # To avoid key error's
             timers = []
             alarms = []
-
-            result = self.get_alarms_and_timers_from(device, API_ENDPOINT_ALARMS)
-            if result:
-                timers = result.get(TIMERS)
-                alarms = result.get(ALARMS)
-
-                if not timers and not alarms:
-                    _LOGGER.error(
-                        "For device {device} - {error}".format(
-                            device=device.device_name, error=API_RETURNED_UNKNOWN
+            if device.ip:
+                result = self.get_alarms_and_timers_from(device, API_ENDPOINT_ALARMS)
+                if result:
+                    timers = result.get(TIMERS)
+                    alarms = result.get(ALARMS)
+
+                    if not timers and not alarms:
+                        _LOGGER.error(
+                            "For device {device} - {error}".format(
+                                device=device.device_name, error=API_RETURNED_UNKNOWN
+                            )
                         )
-                    )
+            else:
+                offline_devices.append(device.device_name)
 
             device.timers = timers
             device.alarms = alarms
             _LOGGER.debug(device)
 
```

Comment:
This should be moved inside the `if device.ip`

Reply:
If this is moved it will unregistered all the devices from HA every time it goes offline

Reply:
Hm, i m a bit confused, but sure. Will need to dive deeper into that later when i can get an access to my setup :) Let's just keep it then

Reply:
Actually, your right, we can move it. I was the one getting confused :laughing: